app/graphql/types/plans/object.rb

```diff
@@ -49,6 +49,21 @@ class Object < Types::BaseObject
       field :is_overridden, Boolean, null: false
       field :subscriptions_count, Integer, null: false
 
+      def entitlements
+        # NOTE: Entitlements don't rely on the overridden plan. Unfortunately, the frontend reuse the same
+        #       component for both plan edit and subscription plan override edit.
+        #       In order to keep work minimal in the frontend for now, return the subscription entitlements instead of
+        #       in the plan entitlements when resolving a plan override. A plan override is a plan with a parent_id.
+        if object.parent_id.present?
+          subscription = object.organization.subscriptions.find_by(plan: object)
+          pps subscription
+          entitlements = ::Entitlement::SubscriptionEntitlement.for_subscription(subscription)
+          ::V1::Entitlement::SubscriptionEntitlementsCollectionSerializer.new(entitlements).serialize_models
```

Comment:
It feels very weird to rely on a serializer in a graphQL type 🤔 

Also, we are doing a lot of DB queries here, do you think we could preload some objects?

Reply:
Maybe it can be an intermediary model (a basic ruby Data class), what do you think?